src/NServiceBus.RavenDB/Internal/RavenDBSynchronizedStorage.cs

```diff
@@ -11,43 +10,29 @@ namespace NServiceBus.RavenDB.Internal
 
     class RavenDBSynchronizedStorage : ISynchronizedStorage
     {
-        ReadOnlySettings settings;
-
-        public RavenDBSynchronizedStorage(ReadOnlySettings settings)
-        {
-            this.settings = settings;
-        }
-
         public Task<CompletableSynchronizedStorageSession> OpenSession(ContextBag contextBag)
         {
             var ownership = GetRavenSession(contextBag);
-
-            return Task.FromResult((CompletableSynchronizedStorageSession)new RavenDBSynchronizedStorageSession(ownership.Session, ownership.Owns));
+            var synchronizedStorageSession = new RavenDBSynchronizedStorageSession(ownership.Session, ownership.Owns);
+            return Task.FromResult((CompletableSynchronizedStorageSession)synchronizedStorageSession);
         }
 
         SessionOwnership GetRavenSession(ContextBag contextBag)
         {
             Func<IAsyncDocumentSession> sessionFunction;
-            settings.TryGet(RavenDbSettingsExtensions.SharedAsyncSessionSettingsKey, out sessionFunction);
+            contextBag.TryGet(out sessionFunction);
             if (sessionFunction != null)
             {
                 return new SessionOwnership(false, sessionFunction());
             }
 
             IAsyncDocumentSession session;
-            if (settings.TryGet(out session))
+            if (contextBag.TryGet(out session))
             {
                 return new SessionOwnership(true, session);
             }
 
-            session = contextBag.Get<IAsyncDocumentSession>();
-
-            if (session == null)
-            {
-                throw new Exception("IAsyncDocumentSession is not configured in the container for the incoming message pipeline.");
-            }
-
-            return new SessionOwnership(true, session);
+            throw new Exception("IAsyncDocumentSession could not be retrieved for the incoming message pipeline.");
```

Comment:
Personal preference would be to reverse the `if (contextBag.TryGet(out session))` and end on a `return` rather than an `throw`. It's not a position I'll defend to the death though. Or even the slightly injured.